src/gluonts/nursery/autogluon_tabular.py

```diff
@@ -59,19 +61,25 @@ def get_prediction_dataframe(series):
     )
     convert_type = {x: "category" for x in df.columns.values[:4]}
     df = df.astype(convert_type)
-    return df
+
+    cache = [None] * (context_length + prediction_length) + list(series.values)
+    col = []
+    total = len(df)
+    for i in range(context_length):
+        col.append('lag' + str(i))
+        df['lag' + str(i)] = cache[i:i + total]
+    return df if train else df[-prediction_length:]
 
 
 class TabularPredictor(Predictor):
-    def __init__(
-        self,
-        ag_model,
-        freq: str,
-        prediction_length: int,
-    ) -> None:
-        self.ag_model = ag_model  # task?
+    def __init__(self, ag_model, freq: str, prediction_length: int, lags: list, use_lag=False) -> None:
```

Comment:
Why do we have the `use_lag` flag here? Maybe the `lags` list can be enough: if empty (which could be the default) then no lags are used

Reply:
I think an empty list of `lags` may have two indicates: 1) no lags are used 2) generate the lags by `get_lags_for_frequency`.